eclass/rocm.eclass

```diff
@@ -231,3 +235,27 @@ check_amdgpu() {
 }
 
 fi
+
+# @FUNCTION: rocm_use_hipcc
+# @USAGE: rocm_use_hipcc
+# @DESCRIPTION:
+# switch active C and C++ compilers to hipcc, clean unsupported flags and setup ROCM_TARGET_LST file.
+rocm_use_hipcc() {
+	# During the configuration stage, CMake tests whether the compiler is able to compile a simple program.
+	# Since CMake checker does not specify --offload-arch=, hipcc enumerates devices using four methods
+	# until it finds at least one device. Last way is by accessing them (via rocminfo).
+	# To prevent potential sandbox violations, we set the ROCM_TARGET_LST variable (which is checked first).
+	local target_lst="${BUILD_DIR}"/gentoo_rocm_target.lst
```

Comment:
Maybe use temp dir ${T} would be better?

```suggestion
	local target_lst="${T}"/gentoo_rocm_target.lst
```

Reply:
Hm, I think BUILD_DIR is better. `gentoo_rocm_target.lst` now resides near `gentoo_common_config.cmake`, `gentoo_rules.cmake` and `gentoo_toolchain.cmake`.

Reply:
Well, changing to `${T}` can avoid the problem of nonexistent `BUILD_DIR`. In the future there may be other build system other than cmake calling hipcc, and their ebuild may not use cmake.eclass and have `BUILD_DIR` set.